packages/medusa/src/api/routes/store/returns/create-return.js

```diff
@@ -0,0 +1,188 @@
+import { MedusaError, Validator } from "medusa-core-utils"
+import { defaultRelations, defaultFields } from "./"
+
+export default async (req, res) => {
+  const schema = Validator.object().keys({
+    order_id: Validator.string().required(),
+    items: Validator.array()
+      .items({
+        item_id: Validator.string().required(),
+        quantity: Validator.number().required(),
+      })
+      .required(),
+    return_shipping: Validator.object()
+      .keys({
+        option_id: Validator.string().optional(),
+      })
+      .optional(),
+  })
+
+  const { value, error } = schema.validate(req.body)
+  if (error) {
+    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
+  }
+
+  const idempotencyKeyService = req.scope.resolve("idempotencyKeyService")
+
+  const headerKey = req.get("Idempotency-Key") || ""
+
+  let idempotencyKey
+  try {
+    idempotencyKey = await idempotencyKeyService.initializeRequest(
+      headerKey,
+      req.method,
+      req.params,
+      req.path
+    )
+  } catch (error) {
+    res.status(409).send("Failed to create idempotency key")
+    return
+  }
+
+  res.setHeader("Access-Control-Expose-Headers", "Idempotency-Key")
+  res.setHeader("Idempotency-Key", idempotencyKey.idempotency_key)
+
+  try {
+    const orderService = req.scope.resolve("orderService")
+    const returnService = req.scope.resolve("returnService")
+    const eventBus = req.scope.resolve("eventBusService")
+
+    let inProgress = true
+    let err = false
+
+    while (inProgress) {
+      switch (idempotencyKey.recovery_point) {
+        case "started": {
+          const { key, error } = await idempotencyKeyService.workStage(
+            idempotencyKey.idempotency_key,
+            async manager => {
+              const order = await orderService
+                .withTransaction(manager)
+                .retrieve(value.order_id, {
+                  select: ["refunded_total", "total"],
+                  relations: ["items"],
+                })
+
+              const returnObj = {
+                order_id: value.order_id,
+                idempotency_key: idempotencyKey.idempotency_key,
+                items: value.items,
+              }
+
+              if (value.return_shipping) {
+                returnObj.shipping_method = value.return_shipping
+              }
+
+              if (typeof value.refund !== "undefined" && value.refund < 0) {
```

Comment:
`refund` is missing in validation schema.

Reply:
Sorry should have removed it completely. The idea here is to always calculate the `refund_amount` as we are now creating the return from the storefront and not through an authenticated admin route.

Reply:
Cool! I should have spotted, that it should be removed - my bad.